hack/windows/Install-OVS.ps1

```diff
@@ -244,49 +423,197 @@ function InstallDependency() {
     sc.exe failure ovsdb-server reset= 0 actions= restart/0/restart/0/restart/0
     Start-Service ovsdb-server
     # Create and start ovs-vswitchd service.
     Log "Create and start ovs-vswitchd service."
-    sc.exe create ovs-vswitchd binpath="$OVSInstallDir\usr\sbin\ovs-vswitchd.exe  --pidfile -vfile:info --log-file  --service" start= auto depend= "ovsdb-server"
+    sc.exe create ovs-vswitchd binpath="$OVSUsrSbinDir\ovs-vswitchd.exe  --pidfile -vfile:info --log-file=$OVSLogDir\ovs-vswitchd.log  --service" start= auto depend= "ovsdb-server"
     sc.exe failure ovs-vswitchd reset= 0 actions= restart/0/restart/0/restart/0
     Start-Service ovs-vswitchd
+
     # Set OVS version.
-    $OVS_VERSION=$(Get-Item $OVSInstallDir\driver\OVSExt.sys).VersionInfo.ProductVersion
+    $OVS_VERSION=$(Get-Item $OVSInstallPath\driver\OVSExt.sys).VersionInfo.ProductVersion
     Log "Set OVS version to: $OVS_VERSION"
-    ovs-vsctl --no-wait set Open_vSwitch . ovs_version=$OVS_VERSION
+    & $OVSUsrBinDir\ovs-vsctl.exe --no-wait set Open_vSwitch . ovs_version=$OVS_VERSION
+
+    # Add OVS usr/bin and usr/sbin to the environment path.
+    AddToEnvPath($OVSUsrSbinDir)
+    AddToEnvPath($OVSUsrBinDir)
+
+    # Antrea Pod runs as NT AUTHORITY\SYSTEM user on Windows, antrea-ovs container writes
+    # PID and conf.db files to $OVSInstallDir on Windows Node when it is running.
+    icacls $OVSInstallPath /grant "NT AUTHORITY\SYSTEM:(OI)(CI)F" /T
 }
 
-Log "Installation log location: $InstallLog"
+function InstallOVSServices() {
+    param (
+        [parameter(Mandatory = $true)] [string] $OVSInstallPath
+    )
+
+    # Remove the existing OVS Services to avoid issues.
+    If (ServiceExists("ovs-vswitchd")) {
+        stop-service ovs-vswitchd
+        sc.exe delete ovs-vswitchd
+    }
+    if (ServiceExists("ovsdb-server")) {
+        stop-service ovsdb-server
+        sc.exe delete ovsdb-server
+    }
+
+    # Install OpenSSL dependencies.
+    $OVSBinPaths="${OVSInstallPath}\usr\bin;${OVSInstallPath}\usr\sbin"
+    InstallOpenSSLFiles "$OVSBinPaths"
+
+    # Install OVS Services and configure OVSDB.
+    ConfigOVS($OVSInstallPath)
+}
+
+function PrepareOVSLocalFiles() {
+    $OVSDownloadURL = $DefaultOVSDownloadURL
+    $desiredOVSPublishedHash = $DefaultOVSPublishedHash
+    if ($LocalFile -ne "") {
+        if (-not (Test-Path $LocalFile)){
+            Log "Path $LocalFile doesn't exist, exit"
+            exit 1
+        }
 
-CheckIfOVSInstalled
+        $ovsFile = Get-Item $LocalFile
+        if ($ovsFile -is [System.IO.DirectoryInfo])  {
+            return $ovsFile.FullName
+        }
+
+        # $ovsFile as a zip file is supported
+        $attributes = $ovsFile.Attributes
+        if (("$attributes" -eq "Archive") -and ($ovsFile.Extension -eq ".zip" ) ) {
```

Comment:
I just realized that the behavior is very different when `LocalFile` is a directory vs an archive file. When it's a directory, we use it as is and when it is an archive, we unzip it to `OVSInstallDir`. I wonder if that makes sense.